shared/definition.go

```diff
@@ -454,23 +454,24 @@ func (d *Definition) GetRunnableActions(trigger string) []DefinitionAction {
 	return out
 }
 
-// GetEarlyPackages returns a list of packages which are to be installed or removed earlier than the actual package handling.
+// GetEarlyPackages returns a list of packages which are to be installed or removed earlier than the actual package handling
+// Also removes them from the package set so they aren't attempted to be re-installed again as normal packages
 func (d *Definition) GetEarlyPackages(action string) []string {
-	var out []string
+	var early []string
 
-	for _, set := range d.Packages.Sets {
-		if set.Action != action || !set.Early {
-			continue
-		}
+	normal := d.Packages.Sets[:0]
```

Comment:
any reason not to use `normal := []shared.DefinitionPackagesSet{}` instead?

Reply:
No, that sounds good to me!